### ISSUE TITLE
Minification / Rollup improvements

### ISSUE THREAD

**Issue by gfwilliams** [espruino/EspruinoWebIDE#242]
Right now the IDE has two options - Esprima and the online closure compiler (which is slow, and rate limited).

@opichals has done some great work getting rollup working with EspruinoTools and the IDE, but I've been really struggling to get this to work reliably on espruino.com/ide. Original PRs here:

https://github.com/espruino/EspruinoWebIDE/pull/197
https://github.com/espruino/EspruinoTools/pull/80

For basically the last 2 years (since the merge) I've been maintaining a private fork that I build espruino.com/ide and the Web app from because I can't get it working nicely, and I just had another go for a few hours and haven't had success. So rather than waiting until I break something permanently in my fork (and having a public repo that doesn't work), I'm just going to revert the PR for now and open this issue.

I still really want to get this going, but I'll be adding a serverside element to the IDE soon and it might make more sense to implement rollup in that - it seems that realistically I may need to rethink how modules are handled in the IDE in order to get it to integrate nicely.

**Comment by opichals** [comment]
@gfwilliams Fair enough. The rollup and used plugins need updating anyway from the original PR shape.

I did that work and released that work but didn't get to update the EspruinoTools and browser support yet.

**Comment by gfwilliams** [comment]
Thanks - I'll have a look into this when I get a moment and will see if I can come up with some kind of system that's flexible enough to integrate cleanly with Rollup.
